File: php/features.php
<?php
declare(strict_types=1);

// GregorianLunarCalendar SDK feature factory

require_once __DIR__ . '/feature/BaseFeature.php';
require_once __DIR__ . '/feature/TestFeature.php';


class GregorianLunarCalendarFeatures
{
    public static function make_feature(string $name)
    {
        switch ($name) {
            case "base":
                return new GregorianLunarCalendarBaseFeature();
            case "test":
                return new GregorianLunarCalendarTestFeature();
            default:
                return new GregorianLunarCalendarBaseFeature();
        }
    }
}

File: py/features.py
# GregorianLunarCalendar SDK feature factory

from feature.base_feature import GregorianLunarCalendarBaseFeature
from feature.test_feature import GregorianLunarCalendarTestFeature


def _make_feature(name):
    features = {
        "base": lambda: GregorianLunarCalendarBaseFeature(),
        "test": lambda: GregorianLunarCalendarTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
